Panther-enhanced fork of Stratus Red Team (#16)
panther-labs/stratus-red-team: README.md

[![made-with-Go](https://img.shields.io/badge/Made%20with-Go-1f425f.svg)](http://golang.org)  [![Tests](https://github.com/DataDog/stratus-red-team/actions/workflows/test.yml/badge.svg)](https://github.com/DataDog/stratus-red-team/actions/workflows/test.yml) [![static analysis](https://github.com/DataDog/stratus-red-team/actions/workflows/static-analysis.yml/badge.svg)](https://github.com/DataDog/stratus-red-team/actions/workflows/static-analysis.yml) ![Maintainer](https://img.shields.io/badge/maintainer-@christophetd-blue) [![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/DataDog/stratus-red-team/badge)](https://api.securityscorecards.dev/projects/github.com/DataDog/stratus-red-team) [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/6530/badge)](https://bestpractices.coreinfrastructure.org/projects/6530)

> Stratus Red Team is "[Atomic Red Team](https://github.com/redcanaryco/atomic-red-team)™" for the cloud, allowing to emulate offensive attack techniques in a granular and self-contained manner.

Reviewer: ```suggestion
## Panther-enhanced fork of Stratus Red Team
[Stratus Red Team](https://github.com/DataDog/stratus-red-team) is an open-source adversary emulation tool that lets you validate the security of your cloud environments by replicating offensive attack techniques in a granular and self-contained manner. It can be thought of as [Atomic Red Team](https://github.com/redcanaryco/atomic-red-team)™ for the cloud. 

This repository is a Panther-enhanced fork of Stratus Red Team that adds **end-to-end integration testing for detection engineering**. Instead of using fabricated test cases that may not match production logs, you can use this fork to generate authentic log data, allowing for better attack simulation and detection validation.
```

Reviewer: My intent here was to say what this repo is sooner—i.e., above the video/gif

  </a>
</p>

## Panther-Enhanced Fork

> This fork adds **end-to-end integration testing for detection engineering**. Instead of crafted test cases that may not match production logs, simulate real attacks to generate authentic log data for detection validation.

Reviewer: ```suggestion
```

Reviewer: moved above

> ### Why End-to-End Testing?

Reviewer: ```suggestion
### Why perform end-to-end testing?
```

Reviewer: In the docs we use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings) in headings, so I encourage that here. If you prefer title-style capitalization for headers, that's fine, just keep one style consistent throughout

Traditional detection testing uses theoretical log samples that may differ from real-world patterns. This approach:
- **Generates authentic logs** from actual attack simulations  
- **Tests the complete pipeline** from log collection to alerting
- **Enables test-driven detection development** with real attack patterns
- **Provides confidence** that idle detection rules will work when attacks occur

> ### Panther Enhancements

Reviewer: ```suggestion
### Panther enhancements
```

> **Custom Prefix Support** - Use `--prefix` to avoid "stratus-red-team" detection by AI systems:

Reviewer: ```suggestion
#### Custom prefix support
Use `--prefix` to avoid "stratus-red-team" detection by AI systems:
```

```bash
stratus detonate aws.persistence.iam-create-admin-user --prefix "security-test"
```

> **New Attack Techniques** - Enhanced AI/Bedrock security testing:

Reviewer: ```suggestion
#### Additional attack techniques
Enhanced AI/Bedrock security testing:
```

Reviewer: I tend to be careful with "New" as it's time-in-place-specific

- `aws.defense-evasion.bedrock-guardrail-delete` - Delete AI safety guardrails
- `aws.defense-evasion.bedrock-model-invocation-logging-delete` - Disrupt AI activity monitoring
- `aws.impact.bedrock-converse-guardrail-trigger` - Test guardrail limits via Converse API
- `aws.impact.bedrock-invoke-model-guardrail-trigger` - Test guardrail limits via InvokeModel API

> - `aws.persistence.iam-create-admin-user` - customized to print access key ID

Reviewer: ```suggestion
- `aws.persistence.iam-create-admin-user` - Customized to print access key ID
```

- `gcp.privilege-escalation.tag-based-privilege-escalation` - Escalate GCP compute privileges via conditional access tags

> **Detection Workflow**:

Reviewer: ```suggestion
#### Detection Workflow
```

> 1. Run attack simulations → Generate authentic logs

Reviewer: ```suggestion
1. Run attack simulations to generate authentic logs
```

> 2. Export CloudTrail/app logs → Create test fixtures  

Reviewer: ```suggestion
2. Export CloudTrail/application logs
3. Create test fixtures using the logs you exported
```

> 3. Write detections → Based on real attack patterns

Reviewer: ```suggestion
4. Write detections
```

> 4. Re-run simulations → Validate end-to-end pipeline

Reviewer: ```suggestion
5. Re-run attack simulations to validate your pipeline end-to-end
```

> ### Related Projects & Talks

Reviewer: ```suggestion
### Related projects and talks
```

> **Sister Project**: [Grimoire](https://github.com/panther-labs/grimoire) - Panther's fork of Grimoire for end-to-end detection testing automation

Reviewer: ```suggestion
#### Grimoire
Panther also has a [fork of Grimoire](https://github.com/panther-labs/grimoire), which you can use for end-to-end detection testing automation
```

> **Talks**:

Reviewer: ```suggestion
#### Talks
```
